src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs

```diff
@@ -572,7 +560,7 @@ protected virtual void PreMempoolChecks(MempoolValidationContext context)
 
             // Coinstake is only valid in a block, not as a loose transaction
             // TODO: mempool needs to have seprate checks for POW/POS as part of the change to rules.
-            if (context.Transaction.IsCoinStake)
+            if (this.network.Consensus.IsProofOfStake && context.Transaction.IsCoinStake)
```

Comment:
This breaks one of the mempool unit tests but I'll fix the test in my open PR.

Reply:
ah really? this is correct though right?

Reply:
Yes; the test was taking a shortcut defining a coinstake on a non-PoS network.